src/util/fetch.ts

```diff
@@ -0,0 +1,18 @@
+import {getCookie , setCookie, deleteCookie} from '../util/Authentication'
+
+export const fetchAPIData = async (url: string, method: string, body:object , headers?:object) => {
+    const cookie = getCookie('shelf-share-session')as { token: string }
+    const token = cookie.token
+    const data = await fetch(url , {
+        method,
+        body: JSON.stringify(body),
+        headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+            ...headers,
+        },
+    })
+    const response = await data.json();
```

Comment:
Technically data and response variable names should be flipped here. Response is what we get back from fetch, whereas data tends to imply we have parsed the response. Can you flip the names here.

After flipping the variable names, can you add in a check before running response.json()? Check response.status (the status code of the response). If it starts with 4 or 5 (so a 400 or 500 error code) then throw an error with the code instead of processing the response.